### PR TITLE
Remove the lib/calabash directory and put Dylibs module in Calabash::Cucumber

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

Resolves **move lib/calabash/ directory; it clobbers the namespace of the calabash/calabash gem** #956

* Moved lib/calabash/dylibs.rb to lib/calabash-cucumber/dylibs.rb
* Changed module to Calabash::Cucumber
* Raise an error if you try to inject a dylib when targeting a physical device

### To Review

```
$ git fetch
$ git co -t origin/feature/move-calabash-directory
$ cd calabash-cucumber
$ be rake unit
$ be rspec spec/integration/launcher/inject_dylib_spec.rb
```

### CODE REVIEW

File: calabash-cucumber/lib/calabash-cucumber/dylibs.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      # @!visibility private
+      def self.path_to_sim_dylib
+        File.join(self.dylib_dir, self.sim_dylib_basename)
+      end
+
+      # @!visibility private
+      def self.device_dylib_basename
+        "libCalabashDyn.dylib"
```
Comment (sapieneptus) [inline calabash-cucumber/lib/calabash-cucumber/dylibs.rb]: Not `FAT` ?

File: calabash-cucumber/spec/integration/launcher/inject_dylibs_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  it "Launch and inject a dylib on a simulator" do
+    launcher.relaunch(options)
+    expect(launcher.run_loop).not_to be == nil
+  end
+
+
+  it "Raises an error if trying to inject on a device target" do
+    options[:device_target] = "83b59716a3ac25e997770a91477ef4e6ad0ab7bb"
```
Comment (sapieneptus) [inline calabash-cucumber/spec/integration/launcher/inject_dylibs_spec.rb]: Is this Denis? Does this work in CI?

File: calabash-cucumber/spec/lib/dylibs_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+describe Calabash::Cucumber::Dylibs do
+
+  before do
+    path = File.expand_path(File.join("tmp", "dylib-tests", "dylibs"))
+    FileUtils.rm_rf(path)
+    FileUtils.mkdir_p(path)
+
+    FileUtils.touch(File.join(path, "libCalabashDynSim.dylib"))
+    FileUtils.touch(File.join(path, "libCalabashDyn.dylib"))
+
```
Comment (sapieneptus) [inline calabash-cucumber/spec/lib/dylibs_spec.rb]: Why not `FAT` ?

**Comment by jmoody** [comment]
`FAT` is not included in the gem.

re: the UDID - that is fake udid.  Everything is mocked.

**Pull request merged**